### PR TITLE
Fix: use __dirname to find transform files (refs #2)

### PR DESCRIPTION
**Pull request by vitorbal**
The executable was not using `__dirname` to look for the transforms before, which was causing it to error out when installed through npm (See #2). This PR should fix the issue.

### CODE REVIEW

**Comment by nzakas** [comment]
LGTM

**Comment by nzakas** [comment]
@vitorbal does this fix #2? If so, can you update the commit message to say "fixes #2" instead of "refs #2"?

**Comment by vitorbal** [comment]
@nzakas the first commit does not fix #2, only part of the problem, that's why I chose to use "refs".

I just pushed a second commit that does indeed fix #2 entirely. It does away with the `shelljs-nodecli` package in favor of the approach described in [your blog post](https://www.nczonline.net/blog/2016/03/mimicking-npm-script-in-node-js/) (thanks again for the link!).

The bin should now be able to find jscodeshift if eslint-transforms is installed globally or locally, and also if dependencies are flattened. Let me know if the code makes sense from your perspective!

File: bin/eslint-transforms.js [hunk 1]
```diff
@@ -2,15 +2,44 @@
...
 
 var argv = process.argv.slice(2);
 var args = argv.slice(1);
 var transform = argv[0];
 
-nodeCLI.exec(
+/**
+ * Add possible node_modules/.bin paths to env and run the command passed in.
+ *
+ * @param {String} cmd The command to run
```
Comment (nzakas) [inline bin/eslint-transforms.js]: {string} (lowercase)

File: bin/eslint-transforms.js [hunk 2]
```diff
@@ -2,15 +2,44 @@
...
+        // Covers case when dependencies are not flattened and the jscodeshift bin can be found under the
+        // node_modules folder of our package
+        path.resolve(__dirname, "../node_modules/.bin"),
+        env.PATH
+    ].join(SEPARATOR);
+
+    execSync(cmd, {
+        env: env,
+        cwd: process.cwd(),
+        stdio: "inherit"
```
Comment (nzakas) [inline bin/eslint-transforms.js]: I'm not sure, but I don't think you need this.

File: bin/eslint-transforms.js [hunk 3]
```diff
@@ -2,15 +2,44 @@
 
 "use strict";
 
-var nodeCLI = require("shelljs-nodecli");
+var execSync = require("child_process").execSync;
+var objectAssign = require("object-assign");
```
Comment (nzakas) [inline bin/eslint-transforms.js]: If you want to target Node 4 and higher, you can just use `Object.assign` directly.

Comment (vitorbal) [inline bin/eslint-transforms.js]: Would it be okay to do that here already? I thought it would perhaps be better if I clean this up when I submit a separate PR for #4

File: bin/eslint-transforms.js (hunk 2 above)
Comment (vitorbal) [inline bin/eslint-transforms.js]: I tried to remove it but then all output is swallowed.

Comment (nzakas) [inline bin/eslint-transforms.js]: Hmm ok. Nevermind. :)

File: bin/eslint-transforms.js (hunk 3 above)
Comment (nzakas) [inline bin/eslint-transforms.js]: Yeah, that's fine. I generally don't like adding code that I intend to remove later (that's how tech debt is created), but since it's mostly up to you already, that's fine. :)

File: bin/eslint-transforms.js (hunk 1 above)
Comment (nzakas) [inline bin/eslint-transforms.js]: I guess this is the only thing left to change?

File: bin/eslint-transforms.js (hunk 3 above)
Comment (vitorbal) [inline bin/eslint-transforms.js]: Okay, duly noted!

File: bin/eslint-transforms.js (hunk 1 above)
Comment (vitorbal) [inline bin/eslint-transforms.js]: Correct! Just pushed a fix for it!
